hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-catalog/hadoop-yarn-applications-catalog-webapp/src/main/java/org/apache/hadoop/yarn/appcatalog/application/YarnServiceClient.java

```diff
@@ -46,6 +47,19 @@
 public class YarnServiceClient {
 
   private static final Logger LOG = LoggerFactory.getLogger(YarnServiceClient.class);
+
+  /**
+   * It is more performant to reuse ObjectMapper instances but keeping the instance
```

Comment:
not 100% sure about that performant: isn't there some synchronisation locking going on so performance with a single mapper across many threads worse than 1 mapper per thread?

Reply:
I am a committer on the Jackson project. There is minimal synchronisation and I am actively removing many of the remaining cases. Even today, it is more performant to reuse ObjectMapper instead of creating multiple instances.

Reply:
are you? never knew that.